tests/e2e/gateway_test.go

```diff
@@ -0,0 +1,113 @@
+
+	. "github.com/onsi/gomega"
+)
+
+const (
+	gatewayName          = "odh-gateway"
+	gatewayClassName     = "odh-gateway-class"
+	gatewayNamespace     = "openshift-ingress"
+	gatewayServiceName   = "default-gateway"
+	gatewayTLSSecretName = "default-gateway-tls"
+)
+
+var (
+	GatewayServiceGVK = schema.GroupVersionKind{
+		Group:   "services.platform.opendatahub.io",
+		Version: "v1alpha1",
+		Kind:    "GatewayConfig",
+	}
+
+	GatewayClassGVK = schema.GroupVersionKind{
+		Group:   "gateway.networking.k8s.io",
+		Version: "v1",
+		Kind:    "GatewayClass",
+	}
+
+	GatewayGVK = schema.GroupVersionKind{
+		Group:   "gateway.networking.k8s.io",
+		Version: "v1",
+		Kind:    "Gateway",
+	}
+
+	SecretGVK = schema.GroupVersionKind{
+		Group:   "",
+		Version: "v1",
+		Kind:    "Secret",
+	}
+)
+
+func gatewayTestSuite(t *testing.T) {
+	t.Helper()
+	ctx, err := NewTestContext(t)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("Gateway infrastructure validation", func(t *testing.T) {
+		validateGatewayCreation(t, ctx)
+	})
+}
+
+func validateGatewayCreation(t *testing.T, ctx *TestContext) {
+	t.Helper()
+
+	t.Log("Validating Gateway service and API resources creation")
+
+	t.Run("GatewayConfig service CR should be created", func(t *testing.T) {
+		ctx.EnsureResourceExists(
+			WithMinimalObject(GatewayServiceGVK, types.NamespacedName{Name: gatewayServiceName}),
+		)
+	})
+
+	t.Run("GatewayClass should be created", func(t *testing.T) {
+		ctx.EnsureResourceExists(
+			WithMinimalObject(GatewayClassGVK, types.NamespacedName{Name: gatewayClassName}),
+		)
+	})
+
+	t.Run("Certificate secret should be created", func(t *testing.T) {
+		ctx.EnsureResourceExists(
+			WithMinimalObject(SecretGVK, types.NamespacedName{
+				Name:      gatewayTLSSecretName,
+				Namespace: gatewayNamespace,
+			}),
+		)
+	})
+
+	t.Run("Gateway should be created with correct configuration", func(t *testing.T) {
+		ctx.EnsureResourceExists(
+			WithMinimalObject(GatewayGVK, types.NamespacedName{
+				Name:      gatewayName,
+				Namespace: gatewayNamespace,
+			}),
+		)
+
+		// Validate Gateway configuration using the test framework's pattern
+		ctx.EnsureResourceExists(
+			WithMinimalObject(GatewayGVK, types.NamespacedName{
+				Name:      gatewayName,
+				Namespace: gatewayNamespace,
+			}),
+			WithCondition(And(
+				jq.Match(`.spec.gatewayClassName == "%s"`, gatewayClassName),
+				jq.Match(`.spec.listeners | length > 0`),
+				jq.Match(`.spec.listeners[] | select(.name == "https") | .protocol == "%s"`, string(gwapiv1.HTTPSProtocolType)),
+				jq.Match(`.spec.listeners[] | select(.name == "https") | .port == 443`),
+				jq.Match(`.spec.listeners[] | select(.name == "https") | .tls.certificateRefs[0].name == "%s"`, gatewayTLSSecretName),
+			)),
+			WithCustomErrorMsg("Gateway should be created with correct HTTPS configuration"),
+		)
+	})
+
+	t.Log("Gateway API resources validation completed successfully")
+}
```

Comment:
Please align with the table-driven approach we have in all our tests.

Reply:
[Here](https://github.com/opendatahub-io/opendatahub-operator/pull/2418/files#diff-5d3c5a95f330afdc586f440940e5e4bce2dba95b7c64585be9467ee4a3b2065e) is a better way we could implement this.